dss/api/search.py

```diff
@@ -1,47 +1,53 @@
-from .. import dss_handler, get_logger
+from .. import dss_handler, get_logger, DSSException
 from ..util.es import ElasticsearchClient
 
 
 @dss_handler
 def find():
     get_logger().debug("Searching for: %s", request.values["query"])
-    # TODO (mbaumann) Use a connection manager
     es_client = ElasticsearchClient.get(get_logger())
     query = json.loads(request.values["query"])
-    response = Search(using=es_client,
-                      index=DSS_ELASTICSEARCH_INDEX_NAME,
-                      doc_type=DSS_ELASTICSEARCH_DOC_TYPE).query("match", **query).execute()
-    return {"query": query, "results": format_results(request, response)}
+    search_obj = Search(using=es_client,
+                        index=DSS_ELASTICSEARCH_INDEX_NAME,
+                        doc_type=DSS_ELASTICSEARCH_DOC_TYPE).query("match", **query)
+    return jsonify({"query": query, "results": format_results(request, search_obj)})
 
 
 @dss_handler
 def post(query: dict):
     get_logger().debug("Received posted query: %s", json.dumps(query, indent=4))
-    # TODO (mbaumann) Use a connection manager
-    es_client = ElasticsearchClient.get(get_logger())
-    response = Search(using=es_client,
-                      index=DSS_ELASTICSEARCH_INDEX_NAME,
-                      doc_type=DSS_ELASTICSEARCH_DOC_TYPE).update_from_dict(query).execute()
-    return {"query": query, "results": format_results(request, response)}
-
-
-def format_results(request, response):
+    try:
+        es_client = ElasticsearchClient.get(get_logger())
+        search_obj = Search(using=es_client,
+                            index=DSS_ELASTICSEARCH_INDEX_NAME,
+                            doc_type=DSS_ELASTICSEARCH_DOC_TYPE).update_from_dict(query)
+        return jsonify({"query": query, "results": format_results(request, search_obj)})
+
+    except ElasticsearchDslException:
+        raise DSSException(requests.codes.bad_request,
+                           "elasticsearch_query_error",
+                           "Invalid query")
+    except ElasticsearchException:
+        raise DSSException(requests.codes.internal_server_error,
+                           "elasticsearch_error",
+                           "Elasticsearch operation failed")
+
+def format_results(request, search_obj):
     # TODO (mbaumann) extract version from the request path instead of hard-coding it here
     # The previous code worked for post but incorrectly included the query string in the case of get.
     bundles_url_base = request.host_url + 'v1/bundles/'
-    result_list = []
-    for hit in response:
-        result = {
-            'bundle_id': hit.meta.id,
-            'bundle_url': bundles_url_base + hit.meta.id,
-            'search_score': hit.meta.score
-        }
-        result_list.append(result)
+    result_list = [{
+        'bundle_id': hit.meta.id,
+        'bundle_url': bundles_url_base + hit.meta.id.replace(".", "?version=", 1),
```

Comment:
I'm not sure what you're asking.The meta.id represents the full bundle Id and the first dot is the separator between the UUID and the version. The dot in the version is retained as you show above.